WriteVHDLSyntax.py

```diff
@@ -261,27 +310,54 @@ def writeMemoryUtil(memDict, memInfoDict):
 
 
             if memInfo.is_binned:
-                if memInfo.downstream_mtype_short in ("TP", "MP") :
-                    varStr = "_64_4b"
-                else:
-                    varStr = "_8_5b"
+                varStr = "_64_4b"
             else:
                 varStr = "_7b"
             tName = "t_"+mtypeB+"_NENT"
-            if combined:
-              if  "VMSME" in tName :
-                ss += "  subtype "+tName+" is std_logic_vector(31 downto 0);\n"
+            if memInfo.is_binned:
+                ss += "  subtype "+tName+" is std_logic_vector(63 downto 0);\n"
                 tName = "t_"+mtypeB+"_NENTADDR"
-                ss += "  subtype "+tName+" is std_logic_vector(4 downto 0);\n"
-              else:
-                ss += "  subtype "+tName+" is t_arr"+str(num_pages)+varStr+";\n"
+                nentaddrbits = "4"
+                if "VMSTE" in mtypeB :
+                    nentaddrbits = "3"
+                ss += "  subtype "+tName+" is std_logic_vector("+nentaddrbits+" downto 0);\n"
+                tName = "t_"+mtypeB+"_NENTADDRDISK"
+                nentaddrbits = "5"
+                if "VMSTE" in mtypeB :
+                    nentaddrbits = "4"
+                ss += "  subtype "+tName+" is std_logic_vector("+nentaddrbits+" downto 0);\n"
             else:
-              ss += "  subtype "+tName+" is t_arr"+str(num_pages)+varStr+";\n"
+                ss += "  subtype "+tName+" is t_arr"+str(num_pages)+varStr+";\n"
             if memInfo.is_binned:
-                if memInfo.downstream_mtype_short in ("TP", "MP") :
-                    varStr = "_64_1b"
-                    tName = "t_"+mtypeB+"_MASK"
-                    ss += "  subtype "+tName+" is t_arr"+str(num_pages)+varStr+";\n"
+                varStr = "_64_1b"
+                tName = "t_"+mtypeB+"_MASK"
+                ss += "  subtype "+tName+" is t_arr"+str(num_pages)+varStr+";\n"
+                tName = "t_"+mtypeB+"_MASK_"+str(num_pages)
+#                ss += "  subtype "+tName+" is t_arr"+str(num_pages)+varStr+";\n"
+                ss += "  subtype "+tName+" is std_logic_vector("+str(num_pages)+"*64-1 downto 0);\n"
+                varStr = "_128_1b"
+                tName = "t_"+mtypeB+"_MASKDISK"
+                ss += "  subtype "+tName+" is t_arr"+str(num_pages)+varStr+";\n"
+                tName = "t_"+mtypeB+"_MASKDISK_"+str(num_pages)
+#                ss += "  subtype "+tName+" is t_arr"+str(num_pages)+varStr+";\n"
+                ss += "  subtype "+tName+" is std_logic_vector("+str(num_pages)+"*128-1 downto 0);\n"
+                vmstubwidth = 17
+                if "16" in mtypeB:
+                    vmstubwidth = 16
```

Comment:
@jasonfan393 Any reason not to simplify this with the following?
```suggestion
                vmstubwidth = memInfo.bitwidth
```